packages/flutter/lib/src/gestures/mouse_tracking.dart

```diff
@@ -0,0 +1,308 @@
+      && globalPosition == other.globalPosition;
+  }
+
+  @override
+  String toString() => '$runtimeType(globalPosition: $globalPosition, sourceTimeStamp: $sourceTimeStamp)';
+}
+
+/// Details object for [MouseEnterCallback]s.
+class MouseEnterDetails extends _MouseDetails {
+  /// Creates details for a [MouseMoveCallback] call.
+  ///
+  /// The [globalPosition] argument must be provided and must not be null.
+  MouseEnterDetails({Duration sourceTimeStamp, Offset globalPosition})
+      : super(sourceTimeStamp: sourceTimeStamp, globalPosition: globalPosition);
+}
+
+/// The `details` object provides the position of the mouse pointer at the
+/// time of entry.
+typedef MouseEnterCallback = void Function(MouseEnterDetails details);
+
+/// Details object for callbacks that use [MouseMoveCallback].
+class MouseMoveDetails extends _MouseDetails {
+  /// Creates details for a [MouseMoveCallback] call.
+  ///
+  /// The [globalPosition] argument must be provided and must not be null.
+  MouseMoveDetails({Duration sourceTimeStamp, Offset globalPosition})
+    : super(sourceTimeStamp: sourceTimeStamp, globalPosition: globalPosition);
+}
+
+/// The `details` object provides the global position of the mouse pointer.
+typedef MouseMoveCallback = void Function(MouseMoveDetails details);
+
+/// Details object for callbacks that use [MouseExitCallback].
+class MouseExitDetails extends _MouseDetails {
+  /// Creates details for a [MouseMoveCallback] call.
+  ///
+  /// The [globalPosition] argument must be provided and must not be null.
+  MouseExitDetails({Duration sourceTimeStamp, Offset globalPosition})
+      : super(sourceTimeStamp: sourceTimeStamp, globalPosition: globalPosition);
+}
+
+/// The `details` object provides the global position of the mouse pointer at
+/// the time of exit.
+typedef MouseExitCallback = void Function(MouseExitDetails details);
+
+/// The annotation object used to annotate layers that are interested in mouse
+/// movements.
+///
+/// This is added to a layer and managed by the [MouseDetector] widget.
+class MouseDetectorAnnotation {
+  /// Creates an annotation that can be used to find layers interested in mouse
+  /// movements.
+  ///
+  /// At least one of the arguments must be non-null.
+  MouseDetectorAnnotation({this.onEnter, this.onMove, this.onExit});
+
+  /// Triggered when a pointer has entered the bounding box of the annotated
+  /// layer.
+  final MouseEnterCallback onEnter;
+
+  /// Triggered when a pointer has moved within the bounding box of the
+  /// annotated layer.
+  final MouseMoveCallback onMove;
+
+  /// Triggered when a pointer has exited the bounding box of the annotated
+  /// layer.
+  final MouseExitCallback onExit;
+
+  @override
+  String toString() {
+    final String none = (onEnter == null && onExit == null && onMove == null) ? '<none>' : '';
+    return '[$runtimeType${hashCode.toRadixString(16)} $none'
+        '${onEnter == null ? '' : 'onEnter '}'
```

Comment:
this will look weird if onEnter is non-null and everything else is null.

I think you just want the strings to have the space on the left instead of the right, and then it'll be fine.